org-code-javabuilder/lib/src/main/java/org/code/javabuilder/LambdaRequestHandler.java

```diff
@@ -73,48 +79,42 @@ public String handleRequest(Map<String, String> lambdaInput, Context context) {
+        new AWSFileWriter(s3Client, outputBucketName, javabuilderSessionId, getOutputUrl);
+    GlobalProtocol.create(
+        outputAdapter, inputAdapter, dashboardHostname, channelId, levelId, fileWriter);
 
-    // Load files to memory and create and invoke the code execution environment
-    try {
-      GlobalProtocol.create(
-          outputAdapter, inputAdapter, dashboardHostname, channelId, levelId, fileWriter);
+    // Create file loader
+    final UserProjectFileLoader userProjectFileLoader =
+        new UserProjectFileLoader(
+            GlobalProtocol.getInstance().generateSourcesUrl(),
+            levelId,
+            dashboardHostname,
+            useNeighborhood);
 
-      try {
-        // Delete any leftover contents of the tmp folder from previous lambda invocations
-        Util.recursivelyClearDirectory(Paths.get(System.getProperty("java.io.tmpdir")));
-      } catch (IOException e) {
-        throw new InternalServerError(INTERNAL_EXCEPTION, e);
-      }
-
-      // Create file loader
-      final UserProjectFileLoader userProjectFileLoader =
-          new UserProjectFileLoader(
-              GlobalProtocol.getInstance().generateSourcesUrl(),
-              levelId,
-              dashboardHostname,
-              useNeighborhood);
-      UserProjectFiles userProjectFiles = userProjectFileLoader.loadFiles();
-      try (CodeBuilder codeBuilder =
-          new CodeBuilder(GlobalProtocol.getInstance(), userProjectFiles)) {
-        codeBuilder.buildUserCode();
-        codeBuilder.runUserCode();
-      }
-    } catch (JavabuilderException | JavabuilderRuntimeException e) {
-      // Send user-facing exceptions to the user and log the stack trace to CloudWatch
-      outputAdapter.sendMessage(e.getExceptionMessage());
-      context.getLogger().log(e.getLoggingString());
-    } catch (InternalFacingException e) {
-      // Send internal-facing exceptions to CloudWatch
-      context.getLogger().log(e.getLoggingString());
-    } catch (Throwable e) {
-      e.printStackTrace();
-    } finally {
-      final DeleteConnectionRequest deleteConnectionRequest =
-          new DeleteConnectionRequest().withConnectionId(connectionId);
-      api.deleteConnection(deleteConnectionRequest);
+    try {
+      // Delete any leftover contents of the tmp folder from previous lambda invocations
+      Util.recursivelyClearDirectory(Paths.get(System.getProperty("java.io.tmpdir")));
+    } catch (IOException e) {
+      outputAdapter.sendMessage(
+          (new InternalServerError(INTERNAL_EXCEPTION, e).getExceptionMessage()));
+      cleanUpResources(connectionId, api);
+      return "error clearing tmpdir";
     }
 
+    // Load files to memory and create and invoke the code execution environment
+    CodeBuilderWrapper codeBuilderWrapper =
+        new CodeBuilderWrapper(userProjectFileLoader, outputAdapter);
+    codeBuilderWrapper.executeCodeBuilder();
+
+    cleanUpResources(connectionId, api);
```

Comment:
Was it intentional to move this outside of the finally block (so this doesn't execute if executeCodeBuilder() throws an exception)?

Reply:
It was, but I'll add it back in. I didn't realize the output adapters had a variety of ways they could throw exceptions.